In [ ]:
import numpy as np
import pandas as pd

In [ ]:
data_profil = pd.read_csv('C:/Users/USER/Downloads/data_profil.csv')
catatan_lokasi = pd.read_csv('C:/Users/USER/Downloads/catatan_lokasi.csv')

In [ ]:
data_profil.head()

id jenis_kelamin                 divisi  umur
0   1     Laki-laki       Customer Service    34
1   2     Laki-laki          Data Engineer    26
2   3     Laki-laki      Software Engineer    28
3   4     Perempuan  Business Intelligence    21
4   5     Perempuan      Software Engineer    21

In [ ]:
catatan_lokasi.head()

id   tanggal   lokasi_dominan
0   1  21/05/19    Jakarta Pusat
1   2  21/05/19  Kabupaten Bogor
2   3  21/05/19    Jakarta Pusat
3   4  21/05/19   Kota Tangerang
4   5  21/05/19       Kota Bogor

In [ ]:
data_gabung = pd.merge(data_profil, catatan_lokasi, on='id', how='right')

In [ ]:
data_gabung

id jenis_kelamin             divisi  umur   tanggal lokasi_dominan
0       1     Laki-laki   Customer Service    34  21/05/19  Jakarta Pusat
1       1     Laki-laki   Customer Service    34  22/05/19  Jakarta Pusat
2       1     Laki-laki   Customer Service    34  23/05/19  Jakarta Pusat
3       1     Laki-laki   Customer Service    34  24/05/19    Kota Bekasi
4       1     Laki-laki   Customer Service    34  25/05/19    Kota Bekasi
...   ...           ...                ...   ...       ...            ...
2595  100     Perempuan  Software Engineer    29  11/06/19  Jakarta Timur
2596  100     Perempuan  Software Engineer    29  12/06/19  Jakarta Timur
2597  100     Perempuan  Software Engineer    29  13/06/19  Jakarta Timur
2598  100     Perempuan  Software Engineer    29  14/06/19  Jakarta Timur
2599  100     Perempuan  Software Engineer    29  15/06/19  Jakarta Timur

[2600 rows x 6 columns]

In [ ]:
data_gabung['tanggal'] = data_gabung['tanggal'].astype(str)
data_gabung['lokasi_dominan'] = data_gabung['lokasi_dominan'].astype(str)
data_gabung['divisi'] = data_gabung['divisi'].astype(str)

## 1. Berapakah umur termuda dari karyawan yang tergabung dalam divisi Data Science?

In [ ]:
data_profil.groupby('divisi').umur.min()

divisi
Business Intelligence    21
Customer Service         22
Data Engineer            23
Data Science             21
Marketing                24
Software Engineer        21
Name: umur, dtype: int64

Dapat dilihat berdasarkan data di atas, maka dapat disimpulkan umur termuda dari karyawan yang tergabung dalam divisi Data Science adalah 21 tahun

## 2. Divisi apa yang karyawannya paling banyak melakukan mudik?

In [ ]:
data_gabung.divisi.value_counts()

Software Engineer        806
Business Intelligence    546
Customer Service         390
Data Engineer            364
Data Science             260
Marketing                234
Name: divisi, dtype: int64

Divisi yang karyawannya paling banyak melakukan mudik adalah Software Engineer yaitu sebanyak 806 karyawan

## 3. Berapakah median umur dari keseluruhan karyawan Tokolapak?

In [ ]:
data_profil['umur'].median()

27.0

Median umur dari keseluruhan karyawan Tokolapak adalah 27

## 4. Tanggal berapakah menjadi hari favorit karyawan Tokolapak pulang ke Jabodetabek?

In [ ]:
data_gabung.tanggal[data_gabung.lokasi_dominan.isin(['Kota Depok',
                                              'Kota Bekasi',
                                              'Jakarta Timur',
                                              'Jakarta Selatan',
                                              'Jakarta Pusat',
                                              'Jakarta Utara',
                                              'Jakarta Barat',
                                              'Kota Bogor',
                                              'Kota Tangerang',
                                              'Kabupaten Bekasi',
                                              'Kota Tangerang Selatan',
                                              'Kabupaten Tangerang',
                                              'Kabupaten Bogor'])].value_counts()

15/06/19    100
14/06/19    100
21/05/19    100
22/05/19    100
25/05/19     98
24/05/19     98
12/06/19     98
11/06/19     98
23/05/19     97
13/06/19     97
27/05/19     95
26/05/19     95
28/05/19     91
10/06/19     81
29/05/19     81
30/05/19     66
09/06/19     66
31/05/19     42
01/06/19     27
02/06/19     17
08/06/19     17
05/06/19      9
06/06/19      9
07/06/19      9
04/06/19      9
03/06/19      7
Name: tanggal, dtype: int64

Dapat dilihat berdasarkan data di atas, hari favorit karyawan Tokolapak pulang ke Jabodetabek yaitu pada tanggal 21 Mei 2019, 22 Mei 2019, 14 Juni 2019, dan 15 Juni 2019

## 5. Berapa jumlah karyawan yang mudik dan tergabung dalam Divisi Data Science?

In [ ]:
data_gabung.divisi.value_counts()

Software Engineer        806
Business Intelligence    546
Customer Service         390
Data Engineer            364
Data Science             260
Marketing                234
Name: divisi, dtype: int64

Maka jumlah karyawan yang mudik dan tergabung dalam Divisi Data Science adalah sebanyak 260 karyawan

## 6. Berapakah umur karyawan yang paling senior dari karyawan yang tergabung dalam divisi Data Engineer?

In [ ]:
data_profil.groupby('divisi').umur.max()

divisi
Business Intelligence    35
Customer Service         35
Data Engineer            35
Data Science             35
Marketing                35
Software Engineer        34
Name: umur, dtype: int64

Umur karyawan yang paling senior dari karyawan yang tergabung dalam divisi Data Engineer adalah 35 tahun

## 7. Berapa jumlah karyawan TokoLapak?

In [ ]:
Iis_id_karyawan = data_gabung.id.unique()
Iis_id_karyawan

array([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
        14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,
        27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,
        40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  52,
        53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,  65,
        66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,  78,
        79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,  91,
        92,  93,  94,  95,  96,  97,  98,  99, 100], dtype=int64)

In [ ]:
len(Iis_id_karyawan)

100

Jumlah karyawan TokoLapak ada sebanyak 100 karyawan

## 8. Kabupaten/Kota manakah yang menjadi tujuan mudik terbanyak?

In [ ]:
data_gabung['lokasi_dominan'].value_counts()

Kota Depok                   177
Kota Bekasi                  165
Jakarta Timur                163
Kota Bogor                   155
Jakarta Barat                155
Jakarta Pusat                145
Kota Tangerang               127
Kabupaten Bekasi             122
Kabupaten Sleman             120
Kota Yogyakarta              117
Kota Tangerang Selatan       116
Jakarta Selatan              105
Jakarta Utara                100
Kabupaten Bogor               98
Kota Medan                    85
Kota Padang                   81
Kabupaten Tangerang           79
Kabupaten Wonogiri            71
Kabupaten Bantul              68
Kabupaten Lampung Selatan     61
Kota Bandar Lampung           58
Kota Bandung                  54
Kota Surabaya                 52
Kota Semarang                 41
Kabupaten Tasikmalaya         35
Kota Manado                   33
Kota Banda Aceh               10
Kabupaten Kebumen              7
Name: lokasi_dominan, dtype: int64

Kota Depok merupakan tujuan mudik terbanyak yaitu sebanyak 177

## 9. Divisi apakah yang semua anggotanya mudik?

In [ ]:
# Belum cukup data untuk menyimpulkan pertanyaan ini, tetapi bisa dibilang semua divisi melakukan mudik

## 10. Kabupaten/Kota manakah yang menjadi tujuan mudik paling sedikit?

In [ ]:
data_gabung['lokasi_dominan'].value_counts()

Kota Depok                   177
Kota Bekasi                  165
Jakarta Timur                163
Kota Bogor                   155
Jakarta Barat                155
Jakarta Pusat                145
Kota Tangerang               127
Kabupaten Bekasi             122
Kabupaten Sleman             120
Kota Yogyakarta              117
Kota Tangerang Selatan       116
Jakarta Selatan              105
Jakarta Utara                100
Kabupaten Bogor               98
Kota Medan                    85
Kota Padang                   81
Kabupaten Tangerang           79
Kabupaten Wonogiri            71
Kabupaten Bantul              68
Kabupaten Lampung Selatan     61
Kota Bandar Lampung           58
Kota Bandung                  54
Kota Surabaya                 52
Kota Semarang                 41
Kabupaten Tasikmalaya         35
Kota Manado                   33
Kota Banda Aceh               10
Kabupaten Kebumen              7
Name: lokasi_dominan, dtype: int64

Kabupaten Kebumen merupakan Kabupaten/Kota yang menjadi tujuan mudik paling sedikit yaitu 7

## 11. Di hari apakah yang menjadi hari favorit (paling banyak) karyawan Tokolapak melakukan mudik? Asumsikan waktu yang dibutuhkan untuk mudik kurang dari 1 hari, dan apabila dia mulai berada di luar Jabodetabek pada hari tersebut, maka hari tersebutlah yang menjadi dominan lokasi seseorang mudik.

In [ ]:
# Datanya belum cukup untuk menjawab spesifik untuk hal ini,
data_gabung['tanggal'].value_counts()

04/06/19    100
26/05/19    100
24/05/19    100
13/06/19    100
07/06/19    100
08/06/19    100
09/06/19    100
27/05/19    100
01/06/19    100
22/05/19    100
29/05/19    100
02/06/19    100
31/05/19    100
23/05/19    100
03/06/19    100
12/06/19    100
25/05/19    100
28/05/19    100
10/06/19    100
21/05/19    100
05/06/19    100
30/05/19    100
15/06/19    100
11/06/19    100
06/06/19    100
14/06/19    100
Name: tanggal, dtype: int64

Jika diasumsikan data benar, maka semua hari adalah hari favorit

## 12. Divisi apakah yang persentase pemudiknya paling sedikit?

In [ ]:
# Kita ketahui dulu divisi yang pemudiknya paling sedikit
data_gabung.divisi.value_counts()

Software Engineer        806
Business Intelligence    546
Customer Service         390
Data Engineer            364
Data Science             260
Marketing                234
Name: divisi, dtype: int64

Berdasarkan data di atas, divisi yang pemudiknya paling sedikit adalah Marketing

In [ ]:
# diasumsikan kita sudah tau divisi mana yang paling sedikit, kemudian kita hitung persantasenya
case = 0
for divisi in data_gabung.divisi:
    if divisi == "Marketing":
        case += 1

hasil = (case/len(data_gabung['id']))*100
print(f'{hasil}%')

9.0%


Sehingga didapat divisi yang persentase pemudiknya paling sedikit adalah Marketing yaitu hanya sebanyak 9%

## 13. Divisi apakah yang jumlah pemudiknya paling sedikit

In [ ]:
data_gabung.divisi.value_counts()

Software Engineer        806
Business Intelligence    546
Customer Service         390
Data Engineer            364
Data Science             260
Marketing                234
Name: divisi, dtype: int64

Divisi yang jumlah pemudiknya paling sedikit adalah Divisi Marketing yaitu sebanyak 234 karyawan